craft_parts/plugins/uv_plugin.py

```diff
@@ -98,8 +98,13 @@ def _get_rewrite_shebangs_commands(self) -> list[str]:
         return []
 
     def _get_create_venv_commands(self) -> list[str]:
+        # Explicitly request a Python version if provided by the plugin, otherwise use the global
+        # parts interpreter.
+        python_ver = (
+            self._get_system_python_interpreter() or "${PARTS_PYTHON_INTERPRETER}"
+        )
         return [
-            f'uv venv --relocatable --allow-existing --python "{self._get_system_python_interpreter()}" "{self._get_venv_directory()}"',
+            f'uv venv --relocatable --allow-existing --python {python_ver} "{self._get_venv_directory()}"',
```

Comment:
I don't think this is doing what you intend it to do.

![image](https://github.com/user-attachments/assets/127d3a08-92b1-4648-b2bb-c7f14cbc137d)

You probably want the equivalent of this if you're using the interpreter environment variable:

```
$ uv venv -p $(python3 --version | cut -d' ' -f2)
Using CPython 3.13.1 interpreter at: /usr/bin/python3.13
Creating virtual environment at: .venv
Activate with: source .venv/bin/activate
```

Reply:
Actually probably what you want is to always ensure you're using the full path...

```suggestion
            f'uv venv --relocatable --allow-existing --python $(which {python_ver}) "{self._get_venv_directory()}"',
```